src/main/java/com/artipie/ResponseMetricsSlice.java

```diff
@@ -87,16 +89,41 @@ public Response response(
      * Report response to metrics.
      *
      * @param rqline Request line.
+     * @param rqheaders Request headers.
      * @param rsstatus Response status.
      */
-    private void report(final String rqline, final RsStatus rsstatus) {
+    private void report(
+        final String rqline,
+        final Iterable<Map.Entry<String, String>> rqheaders,
+        final RsStatus rsstatus
+    ) {
         if (rsstatus.code().matches("^[45].+$")) {
```

Comment:
@olegmoz what do you think that we extract this check to a new method `RsStatus.isError()` or to some wrapper class to avoid code duplication through the codebase.

Reply:
@Vatavuk nice idea, added an issue to add such method to `RsStatus` and a puzzle to use it here